src/app/courses-page/courses-page/course/course.component.ts

```diff
@@ -1,20 +1,17 @@
-import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Component, Input, Output, EventEmitter } from '@angular/core';
 import { Course } from '../../course';
+import { ChangeDetectionStrategy } from '@angular/compiler/src/core';
 
 @Component({
   selector: 'app-course',
   templateUrl: './course.component.html',
-  styleUrls: ['./course.component.css']
+  styleUrls: ['./course.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
From the first point in task: "Make courses list component smart to handle all the logic." and link that they provided, I think that you need to create three components instead of two. E.g. smart component which will load data, and pass it to second presentation component which will display list, and the last one is course component per each item in array.